src/repositories/questionRepository.ts

```diff
@@ -32,4 +34,26 @@ export class QuestionRepository {
             return questions;
         });  
     }
+
+    findByID(questionID: number): Promise<Question | undefined> {
+        return this.dbClient.query("SELECT QUESTIONS.*, USERS.EMAIL FROM QUESTIONS INNER JOIN USERS ON QUESTIONS.USER_ID = USERS.ID WHERE QUESTIONS.ID = ?", questionID)
+        .then((results) => {
+            if (results.length > 0) {
+                const result = results[0];
+                const user = new User({ id: result.USER_ID, email: result.EMAIL })
+                const question = new Question({ title: result.TITLE, description: result.DESCRIPTION, createdAt: result.CREATED_AT, user: user, id: result.ID });
+                return question;
+            } 
+        });  
+    }
+
+    getByID(questionID: number): Promise<Question> {
+        return this.findByID(questionID).then((question) => {
+            if (question) {
+                return question;
+            } else {
+                throw new BusinessError("question not found!")
```

Comment:
BusinessError is converted to 400 (Bad request), and for repository scope, it doesn't exist. getByID should throw RecordNotFoundError when an item doesn't exist.